content/articles/speech-to-text-transcription-model-using-deep-speech/index.md

```diff
@@ -0,0 +1,273 @@
+
+The function also has a `while` loop. The `while` loop is used to loop through the audio file. During each loop, it breaks the audio file into the required size (8196 bytes)
+
+The `feedAudioContent` method feeds the audio chunks into the model for transcription. `intermediateDecode()` will decode the model output and produce the corresponding text. Finally, the function will return the results and print the transcribed text.
+
+Before we call this function, we need to download a sample audio file that the model will use.
+
+### Downloading an audio file
+To download the audio file, use the following command:
+
+```bash
+!wget -O speech.wav https://github.com/EN10/DeepSpeech/blob/master/man1_wb.wav?raw=true
+```
+To see if the audio file has been added to our working directory, use this command:
+
+```bash
+!ls
+```
+The output is shown below:
+
+![Dowloaded audio](/engineering-education/speech-to-text-transcription-model-using-deep-speech/added-audio.jpg)
+
+From the image above,  the working directory has two models (acoustic and language model). It also shows the downloaded audio file saved as `speech.wav`.
+### Playing the audio file
+To play the audio file in Google Colab, use this code:
+
+```python
+Audio('speech.wav')
+```
+When we run this code, it will provide an interface that allows you to play the audio in Google Colab. The interface is shown below:
+
+![Audio interface](/engineering-education/speech-to-text-transcription-model-using-deep-speech/audio-interface.jpg)
+
+The interface has a play button that we can press and listen to audio. We will pass `speech.wav` as an argument when we call the `real_time_transcription` function.
+### Calling the function
+To call the `real_time_transcription` function, run this code:
+
+```python
+real_time_transcription('speech.wav')
+```
+This code will produce a text transcript in real-time. The output is shown below:
+
+![Text transcription output](/engineering-education/speech-to-text-transcription-model-using-deep-speech/text-transcription.jpg)
+
+From the output above we can see the model was able to do real-time transcription. The model produced an output bit-by-bit as the audio processing continued. The acoustic and language models produce an accurate transcription.
+
+### Conclusion
+In this tutorial, we have learned how to do a streaming transcription using Deep Speech. We discussed the types of transcription models. We explored the types of pre-trained Deep Speech models. We use the acoustic and language models to build a model that was able to do real-time transcription.
+
+To access this notebook, click [here](https://colab.research.google.com/drive/15amwA8CdN9AApOoZGnYS3roNMxr6mAcy?usp=sharing)
```

Comment:
Ensure this is the correct Colab notebook. The linked notebook is on anomaly detection and not transcription